SmartDeviceLink/SDLFileManager.m

```diff
@@ -526,11 +524,28 @@ - (float)sdl_uploadPercentage:(float)totalBytes uploadedBytes:(float)uploadedByt
  *  @return                 True if the file still needs to be (re)sent to Core; false if not.
  */
 - (BOOL)sdl_canFileBeUploadedAgain:(nullable SDLFile *)file maxUploadCount:(UInt8)maxUploadCount failedFileUploadsCount:(NSMutableDictionary<SDLFileName *, NSNumber<SDLUInt> *> *)failedFileUploadsCount {
-    if (!file || [self hasUploadedFile:file]) {
+    if (![self.currentState isEqualToString:SDLFileManagerStateReady]) {
+        SDLLogE(@"File named %@ failed to upload. The file manager has shutdown so the file will not be sent again.", file.name);
+        return NO;
+    }
+
+    if (!file) {
+        SDLLogE(@"File named %@ can not be uploaded because it is not a valid file.", file.name);
+        return NO;
+    }
+
+    if ([self hasUploadedFile:file]) {
+        SDLLogE(@"File named %@ has already been uploaded.", file.name);
```

Comment:
This should be a debug message, maybe a warning at the most.

Reply:
Fixed